### DISCUSSION TITLE
We need a way to revalidate static routes❗

### DISCUSSION THREAD

**Discussion by Sammeeey** [vercel/next.js#60641] [App Router]
I almost became insane when I tried to revalidate my static page from a server action after adding new records to my database via web form on Vercel.
Locally it works fine.
But in production on Vercel it doesn't.
I was 100% sure this would and should work.
But it seems that Next.js/Vercel simply doesn't bother revalidating static routes via `revalidatePath()` - at least if you don't directly redirect to them afterwards.

That's the server action:
```js
export async function createAd(formPath, prevState, formData){
    let newAd;
    // ...

    try {
        newAd = await Ad.create({
            // ...
        })
    } catch (error) {
        // ...
    }

    revalidatePath('/ad') // <-- static route, new DB record not displayed🛑
    revalidatePath(`/ad/${category}`) // <-- dynamic route, new DB record displayed✔
    redirect(`/ad/${category}/${newAd._id}`)

    return newState
}
```

`/ad` is the static route.
This route doesn't display the newly created ad afterwards.
However it works perfectly fine for the the dynamic route (`/ad/${category}`).

And when I make the route dynamic by adding `export const dynamic = 'force-dynamic'` to the respective `page.tsx`, `revalidatePath()` in my server action suddenly works for that route too.

Shouldn't it actually be possible to revalidate static paths as well?

**Comment by nirvikpurkait** [comment]
Are you facing the issue only on production? In development everything is fine?

if it is yes, then maybe the reason is---

1. For static route the page generates on build time and not request time. The data stays the same even you have changed the data, the updated data does not reflect.
2. If you make it `export const dynamic = 'force-dynamic'`, the page generates on request time. As you are revalidating the page with `revalidatePath()`, that means you are requesting again, as it generates on request now if you have the new data, that gets reflected.

**Reply by Sammeeey** [reply]
Yes, it's only in production.
1. and 2. are both true.

**Comment by rajeshdavidbabu** [comment]
Just hit this issue today
